### ISSUE TITLE
support --ssl-peer-fingerprint for self-signed winrm https hosts

### ISSUE THREAD

**Issue by hh** [chef/knife-windows#298]
self signed host are just that... self signed.
The certificate doesn't mean much, but verifying the fingerprint is just as good.

```
$ knife winrm --winrm-port 5986 --winrm-transport ssl --winrm-password 'FOOBARBAZ' --ssl-peer-fingerprint 758C4D9AC3E77F518529443D659E1260F5720400 -m 10.113.70.228 'winrm e winrm/config/listener'
10.113.70.228 Listener
10.113.70.228     Address = *
10.113.70.228     Transport = HTTPS
10.113.70.228     Port = 5986
10.113.70.228     Hostname
10.113.70.228     Enabled = true
10.113.70.228     URLPrefix = wsman
10.113.70.228     CertificateThumbprint = 758C4D9AC3E77F518529443D659E1260F5720400
10.113.70.228     ListeningOn = 10.113.70.228, 127.0.0.1, ::1, 2001:0:9d38:90d7:2022:2eca:f58e:b91b, fe80::5efe:10.113.70.228%14, fe80::14d9:e7a5:7df7:f42e%15, fe80::2022:2eca:f58e:b91b%13
10.113.70.228 
```

**Comment by mwrock** [comment]
Commented at https://github.com/WinRb/WinRM/issues/151#issuecomment-145458056

**Comment by hh** [comment]
I think we are good as soon as WinRb\WinRM#151 is merged.

**Comment by hh** [comment]
#313 should close this, feedback welcome

**Comment by hh** [comment]
#344 is the new #313 updated to be a single commit against master

**Comment by hh** [comment]
It looks like #344 is ready to merge!

**Comment by hh** [comment]
@chef/client-windows @mwrock Do we look good here now?

**Comment by mwrock** [comment]
We have another minor version bump coming up and I want to get this included. Just need another maintainer +1 which I'll try to secure this week.

**Comment by smurawski** [comment]
Resolved by #350
